src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -1111,6 +1118,13 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  getSegmentDuration(): number {
+    if (this._videoElement.buffered.length) {
+      return (this._videoElement.buffered.end(0) - this._videoElement.buffered.start(0)) / 3;
```

Comment:
1. use `this._videoElement.buffered.length - 1` instead of 0. like we do in `_getLiveEdge`.
2. this calculation is right for both dvr and not? what is the logic of this?